### PR TITLE
Feature/gl batch addition

### PR DESCRIPTION
**Pull request by fivetran-joemarkiewicz**
## PR Overview
**This PR will address the following Issue/Feature:** #13 and [Issue #20](https://github.com/fivetran/dbt_sage_intacct/issues/20) (dbt_sage_intacct)

**This PR will result in the following new package version:** `v0.3.0`

This update is removing the `_fivetran_deleted` field from the `gl_detail` staging models. While this field has been deprecated and is no longer synced via the connector. We should still mark this as breaking since this update could potentially cause failures for customers who are currently leveraging this field.

**Please provide the finalized CHANGELOG entry which details the relevant changes included in this PR:**

### 🚨 Breaking Changes 🚨:
- Removal of the `_fivetran_deleted` field from the `stg_sage_intacct__gl_detail` table due to this field being deprecated within the connector. The relevant information is now available within the `gl_batch` source table.

### Bug Fixes
- Cast the `class_id` field within the following models as `{{ dbt.type_string() }}` in order to ensure datatype consistency for the field across models. This also ensures downstream transformation logic within `dbt_sage_intacct` compiles successfully.
    - `stg_sage_intacct__ap_bill_item`
    - `stg_sage_intacct__ar_invoice_item`

### Feature Updates
- Addition of the `gl_batch` source and relevant downstream models:
    - `stg_sage_intacct__gl_batch_tmp`
    - `stg_sage_intacct__gl_batch`

### Under the Hood
- Updated Maintainer PR Template
- Included auto-releaser GitHub Actions workflow to automate future releases

### Basic Validation
Please acknowledge that you have successfully performed the following commands locally:
- [X] dbt run –full-refresh && dbt test
- [X] dbt run (if incremental models are present) && dbt test

Before marking this PR as "ready for review" the following have been applied:
- [X] The appropriate issue has been linked, tagged, and properly assigned.
- [X] All necessary documentation and version upgrades have been applied. Don't forget to update the version in the README if necessary.
- [x] docs were regenerated (unless this PR does not include any code or yml updates).
- [X] BuildKite integration tests are passing.
- [X] Detailed validation steps have been provided below.

### Detailed Validation
Please share any and all of your validation steps:

All validations were performed within the Sage Intacct Validations Hex workbook. Please refer to the Height ticket for a direct link to these validations. The Hex notebook includes the following validations:
- _fivetran_deleted proper removal from `gl_detail` model
- Proper materialization of the `gl_batch` models

### If you had to summarize this PR in an emoji, which would it be?
 
🍡

### CODE REVIEW

**Review by fivetran-catfritz — CHANGES_REQUESTED** [review]
Ran and tested locally, just one small change needed in the stg yml. Otherwise it all looks in order!

File: .github/PULL_REQUEST_TEMPLATE/maintainer_pull_request_template.md [hunk 1]
```diff
@@ -4,48 +4,26 @@
...
-- [ ] dbt run –vars (if applicable)
+- [ ] dbt run –full-refresh && dbt test
+- [ ] dbt run (if incremental models are present)
 
 Before marking this PR as "ready for review" the following have been applied:
-- [ ] The appropriate issue has been linked and tagged
-- [ ] You are assigned to the corresponding issue and this PR
-- [ ] BuildKite integration tests are passing
+- [ ] The appropriate issue has been linked, tagged, and properly assigned.
+- [ ] All necessary documentation and version upgrades have been applied. Don't forget to update the version in the README if necessary.
```
[the remark this answers is not in the document]
Comment (fivetran-joemarkiewicz) [reply .github/PULL_REQUEST_TEMPLATE/maintainer_pull_request_template.md:18]: Good reminder, updated.

File: models/stg_sage_intacct.yml [hunk 2]
```diff
@@ -412,4 +412,70 @@ models:
...
+        description: ID of the user who last modified the GL batch
+      - name: module
+        description: Module reference of the GL batch
+      - name: pr_batch_key
+        description: PR batch key related to the GL batch
+      - name: record_no
+        description: Individual record number of the GL batch. This is the lookup key for the GL Detail table
+        tests:
+          - unique
+          - not_null
```
[the remark this answers is not in the document]
Comment (fivetran-joemarkiewicz) [reply models/stg_sage_intacct.yml:459]: Good callout! I also was unable to find in the ERD what the unique identifier was. However, I was able to find in the connector code itself that the primary key was in fact `recordno` and in the connector code we test for uniqueness on this field. See more details in the height ticket.

[the remark this answers is not in the document]
Comment (fivetran-catfritz) [reply models/stg_sage_intacct.yml:458]: Very interesting and good method to know. Thanks!

**Review by fivetran-avinash — APPROVED** [review]
All good!

**Pull request merged**